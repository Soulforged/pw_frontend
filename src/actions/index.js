//@flow
import { createActions } from "redux-actions";
import { CALL_API } from "src/constants";
import type { UserInfo, AppError } from "src/types";

const ENTITIES = "@@entities";
const SAVE = "SAVE";
const REQ = "REQUEST";
const SUCCESS = "SUCCESS";
const FAIL = "FAILURE";
const LOGIN_SUCCESS = "LOGIN_SUCCESS";
const LOGIN_REQUEST = "LOGIN_REQUEST";
const LOGIN_FAILURE = "LOGIN_FAILURE";

const saveTypes = (actionName) => {
  const normActionName = actionName.toUpperCase();
  return [
    `${ENTITIES}/${SAVE}_${normActionName}_${REQ}`,
    `${ENTITIES}/${SAVE}_${normActionName}_${SUCCESS}`,
    `${ENTITIES}/${SAVE}_${normActionName}_${FAIL}`
  ];
};

const fetchTypes = actionName => [
  `${actionName}_${REQ}`,
  `${actionName}_${SUCCESS}`,
  `${actionName}_${FAIL}`
];

const fetchEntitiesTypes = actionName => fetchTypes(`${ENTITIES}/${actionName.toUpperCase()}`);

const login = ({ username, password }: UserInfo) => ({
  [CALL_API]: {
    types: [LOGIN_REQUEST, LOGIN_SUCCESS, LOGIN_FAILURE],
    endpoint: "/authenticate/generate-token",
    body: { user: username, password }
  }
});

const syncActions = createActions({
  SET_ERROR: (error: AppError) => error,
  LOGOUT: () => ({}),
  TOGGLE_THEME: () => ({}),
  TOGGLE_MENU: () => ({}),
  TOGGLE_USER_MENU: () => ({}),
});

export default {
  ENTITIES,
  SAVE,
  REQ,
  SUCCESS,
  FAIL,
  saveTypes,
  fetchTypes,
  fetchEntitiesTypes,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGIN_REQUEST,
  login,
  ...syncActions
};
