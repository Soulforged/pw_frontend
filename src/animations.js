//@flow

export const slideTransition = (duration, key, initialValue, endValue, state) => {
  const transitions = {
    entering: { [key]: initialValue, display: "block", opacity: 0 },
    entered: { [key]: endValue, display: "block", opacity: 1 },
  };
  return {
    transition: `opacity ${key} ${duration}ms ease-in-out`,
    [key]: initialValue,
    display: "none",
    opacity: 0,
    ...transitions[state]
  };
};

export const shirnkTransition = (duration, key, initialValue, endValue, state) => {
  const transitions = {
    entering: { [key]: initialValue },
    entered: { [key]: endValue },
  };
  return {
    transition: `${key} ${duration}ms ease-in-out`,
    [key]: initialValue,
    overflow: "hidden",
    ...transitions[state]
  };
};
