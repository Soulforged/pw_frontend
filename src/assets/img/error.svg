<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 80 80"><defs><style>.cls-1{fill:#d0021b;}.cls-2{fill:#fff;}</style></defs><title>Artboard 4 copy</title><circle class="cls-1" cx="40" cy="40" r="40"/><path class="cls-2" d="M42.83,40,57.69,25.14a2,2,0,0,0-2.83-2.83L40,37.17,25.14,22.31a2,2,0,0,0-2.83,2.83L37.17,40,22.31,54.86a2,2,0,1,0,2.83,2.83L40,42.83,54.86,57.69a2,2,0,0,0,1.42.59,2,2,0,0,0,1.41-3.42Z"/></svg>