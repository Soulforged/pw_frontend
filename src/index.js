import React from "react";
import ReactDOM from "react-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import "font-awesome/css/font-awesome.css";
import "react-table/react-table.css";
import "src/index.css";
import App from "src/App";
import registerServiceWorker from "src/registerServiceWorker";

ReactDOM.render(<App />, document.getElementById("root"));
registerServiceWorker();
