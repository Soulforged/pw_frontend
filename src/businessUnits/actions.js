//@flow
import { push } from "react-router-redux";
import { CALL_API } from "src/constants";
import actions from "src/actions";
import { businessUnit, list, error } from "./schemas";

const { fetchEntitiesTypes, saveTypes } = actions;

const BUSINESSUNIT = "BUSINESSUNIT";
const BUSINESSUNITS = "BUSINESSUNITS";

const restPath = body => (body.id ? `/${body.id}` : "");

export const saveBusinessUnit = (body: Object) => ({
  [CALL_API]: {
    types: saveTypes(BUSINESSUNIT),
    endpoint: `/user/businessunit${restPath(body)}`,
    body,
    update: body.id != null,
    key: "businessUnits",
    errorSchema: error,
    after: dispatch => dispatch(push("/business_units"))
  }
});

const fetchAllBusinessUnits = () => ({
  [CALL_API]: {
    types: fetchEntitiesTypes(BUSINESSUNITS),
    endpoint: `/user/businessunit?companyId=1&companyType=Merchant`,
    schema: list(businessUnit),
    errorSchema: error,
    key: "businessUnits",
    invalidatesCache: true
  }
});

export const fetchBusinessUnits = ({ name }: { name: string } = {}) => {
  if (!name) {
    return fetchAllBusinessUnits();
  }
  return ({
    [CALL_API]: {
      types: fetchEntitiesTypes(BUSINESSUNITS),
      endpoint: `/user/businessunit?companyId=1&companyType=Merchant&name=${name}`,
      schema: list(businessUnit),
      errorSchema: error,
      key: "businessUnits",
      invalidatesCache: true
    }
  });
};

export const getBusinessUnit = (id: number) => ({
  [CALL_API]: {
    types: fetchEntitiesTypes(BUSINESSUNIT),
    endpoint: `/user/businessunit/${id}`,
    schema: businessUnit,
    errorSchema: error,
    key: "businessUnits"
  }
});
