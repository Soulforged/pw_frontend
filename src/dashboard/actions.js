//@flow
import { CALL_API } from "src/constants";
import moment from "moment";

export const DASHBOARD_SUMMARY_SUCCESS = "DASHBOARD_SUMMARY_SUCCESS";
export const DASHBOARD_TRENDS_BY_DATE_SUCCESS = "DASHBOARD_TRENDS_BY_DATE_SUCCESS";
export const DASHBOARD_TRENDS_BY_INST_SUCCESS = "DASHBOARD_TRENDS_BY_INST_SUCCESS";
export const DASHBOARD_ERRORS_SUCCESS = "DASHBOARD_ERRORS_SUCCESS";

const now = moment();
const dateTo = now.format("YYYY-MM-DD");
const dateFrom = now.subtract(7, "days").format("YYYY-MM-DD");

export const fetchSummary = (params = { dateFrom, dateTo }) => ({
  [CALL_API]: {
    types: ["DASHBOARD_SUMMARY_REQUEST", DASHBOARD_SUMMARY_SUCCESS, "DASHBOARD_SUMMARY_FAILURE"],
    endpoint: `/bi/totals/transactions?date_from=${params.dateFrom}&date_to=${params.dateTo}&distinct=sender_account_number`
  }
});

export const fetchTrendsByDate = (params = { dateFrom, dateTo }) => ({
  [CALL_API]: {
    types: ["DASHBOARD_TRENDS_BY_DATE_REQUEST", DASHBOARD_TRENDS_BY_DATE_SUCCESS, "DASHBOARD_TRENDS_BY_DATE_FAILURE"],
    endpoint: `/bi/totals/transactions?date_from=${params.dateFrom}&date_to=${params.dateTo}&group_by=date`
  }
});

export const fetchTrendsByInstitution = (params = { dateFrom, dateTo }) => ({
  [CALL_API]: {
    types: ["DASHBOARD_TRENDS_BY_INST_REQUEST", DASHBOARD_TRENDS_BY_INST_SUCCESS, "DASHBOARD_TRENDS_BY_INST_FAILURE"],
    endpoint: `/bi/totals/transactions?date_from=${params.dateFrom}&date_to=${params.dateTo}&group_by=institution_name`
  }
});

export const fetchErrors = (params = { dateFrom, dateTo }) => ({
  [CALL_API]: {
    types: ["DASHBOARD_ERRORS_REQUEST", DASHBOARD_ERRORS_SUCCESS, "DASHBOARD_ERRORS_FAILURE"],
    endpoint: `/bi/totals/transactions?date_from=${params.dateFrom}&date_to=${params.dateTo}&group_by=status&status=ERROR`
  }
});
