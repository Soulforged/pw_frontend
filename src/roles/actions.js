//@flow
import { push } from "react-router-redux";
import { CALL_API } from "src/constants";
import actions from "src/actions";
import { role, list, error } from "./schemas";

const { fetchEntitiesTypes, saveTypes } = actions;

const ROLE = "ROLE";
const ROLES = "ROLE";

const restPath = body => (body.id ? `/${body.id}` : "");

export const saveRole = (body: Object) => ({
  [CALL_API]: {
    types: saveTypes(ROLE),
    endpoint: `/user/role${restPath(body)}`,
    body,
    update: body.id != null,
    key: "roles",
    errorSchema: error,
    after: dispatch => dispatch(push("/roles"))
  }
});

export const fetchRole = id => ({
  [CALL_API]: {
    types: fetchEntitiesTypes(ROLE),
    endpoint: `/user/role/${id}`,
    schema: role,
    errorSchema: error,
    key: "roles"
  }
});

export const fetchRoles = () => ({
  [CALL_API]: {
    types: fetchEntitiesTypes(ROLES),
    endpoint: "/user/role",
    schema: list(role),
    errorSchema: error,
    key: "roles"
  }
});
