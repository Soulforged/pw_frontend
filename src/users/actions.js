//@flow
import { CALL_API } from "src/constants";
import mocks from "src/mocks";
import { user, list, error } from "./schemas";

export const fetchUsers = () => ({
  [CALL_API]: {
    types: ["@@entities/USERS_REQUEST", "@@entities/USERS_SUCCESS", "@@entities/USERS_FAILURE"],
    endpoint: "/user/backofficeuser",
    schema: list(user),
    errorSchema: error,
    key: "users",
  }
});

export const fetchUserByCriteria = ({ criteria, filter }) => {
  if (!filter) {
    return fetchUsers();
  }
  return {
    [CALL_API]: {
      types: ["@@entities/USERS_FILTER_REQUEST", "@@entities/USERS_FILTER_SUCCESS", "@@entities/USERS_FILTER_FAILURE"],
      endpoint: `/user/backofficeuser/${criteria}/${filter}`,
      schema: user,
      errorSchema: error,
      key: "users",
      mock: mocks.users
    }
  };
};
