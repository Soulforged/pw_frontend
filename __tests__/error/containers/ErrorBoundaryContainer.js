//@flow

import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import createStore from "src/reduxconf";

import actions from "src/actions";
import { ErrorBoundary } from "src/error";

const { setError } = actions;
const store = createStore();

const component = () => (
  <Provider store={store} >
    <ErrorBoundary>
      <div />
    </ErrorBoundary>
  </Provider>
);

it("renders succesfully", () => {
  const rendered = renderer.create(component()).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("displays appropriate view on error", () => {
  store.dispatch(setError(new Error("message")));
  const rendered = renderer.create(component()).toJSON();
  expect(rendered).toMatchSnapshot();
});
