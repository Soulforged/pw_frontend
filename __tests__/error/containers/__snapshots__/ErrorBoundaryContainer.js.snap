// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays appropriate view on error 1`] = `
<div>
  <div />
</div>
`;

exports[`renders succesfully 1`] = `
<div>
  <div />
</div>
`;
